### DISCUSSION TITLE
Flutter - signUpWithPhone method returns null as the response

### DISCUSSION THREAD

**Discussion by sumchans** [supabase/supabase#2833] [Questions]
Hello,

I am trying to sign up a user using the phone authentication method, it works without any issues, now how do I get the response data, mainly the values for the id, phone, created_at fields. 

I tried the code below and then printed response.data, what I get is just null, but I do see a user created in supabase.
```
final response = await supabase.auth.signUpWithPhone(phoneNumber, '123456');
    print(response.data);

final response = await supabase.auth.verifyOTP(phoneNumber, otpCode);
    print(response.data);
```

**Comment by dshukertjr** [comment]
Hi @sumchans 

There seems to be a bug on the gotrue server where `user` object is not returned upon successful OTP verification. https://github.com/supabase/gotrue/issues/141

To bypass this issue, you can simply call `signin()` right after your verification for now until the bug gets fixed!

```dart
final response = await supabase.auth.signUpWithPhone(phoneNumber, '123456');

final response = await supabase.auth.verifyOTP(phoneNumber, otpCode);

final response = await supabase.auth.signin(phone: phoneNumber, password:  '123456');
```

**Reply by sumchans** [reply]
Wouldn't the call to sign in send the request for another otp code?

**Reply by dshukertjr** [reply]
@sumchans 
Actually no it won't. OTP is required every single time if you are sining in with only phone number like this:
```dart
supabase.auth.signin(phone: phoneNumber);
```
but only required for sign up for phone number + password sign in. It's like verification email in email sign in.

**Reply by sumchans** [reply]
Does this mean that I have to ask for a password from the user as well? I was planning on a solution with just the phone number and otp code.

**Reply by sumchans** [reply]
It does look like you cant do signIn with phoneNumber and password, its says 'Unsupported email provider'. When signing in it works just with the phone number and then the opt code. Not sure if I am missing something here.

Also when doing signIn I don't get the response data back, it still says null.

**Reply by dshukertjr** [reply]
Hey @sumchans!

>Does this mean that I have to ask for a password from the user as well? I was planning on a solution with just the phone number and otp code.

No, you should only have to ask for the password once. 
When you verify the OTP, you literally run these two codes consecutively.
```dart
final response = await supabase.auth.verifyOTP(phoneNumber, otpCode);

final response = await supabase.auth.signin(phone: phoneNumber, password:  '123456');
``` 
Right before you run these, you should already have asked for the password since the user has just signed up. You should keep that password in a variable until the user verifies the OTP. 

>It does look like you cant do signIn with phoneNumber and password, its says 'Unsupported email provider'. When signing in it works just with the phone number and then the opt code. Not sure if I am missing something here.

Is there a code that I can take a look at? When you run the following after you have confirm the OTP, you should be able to sign in normally. 
```dart
final response = await supabase.auth.signin(phone: phoneNumber, password:  '123456');
```

When something does not work, it always helps us if you could provide the code you are running!

**Reply by sumchans** [reply]
Thanks Tyler for the detailed explanation. 

I was just wondering why should I ask the user for a password. Usually with phone authentication you just verify the user using OTP. That is the way I have been seeing everywhere.

**Reply by dshukertjr** [reply]
@sumchans Yup, you are correct. Generally phone auth happens without passwords. Supabase provides both options, using phone auth with password and without password. 

Your code was using password, so I just assumed you wanted to sign in using password at the moment, but if you want to sign users in without passwords, you can just pass only the phone number to the sign in method like this:

```dart
final res = await supabase.signin(phone: '[User's Phone Number]');
```
This will trigger OTP, so you verify it and the user is signed in!

Note that because of [this](https://github.com/supabase/gotrue/issues/141) bug on gotrue's server side code, you will not be able to receive the user object when you login using just phone number. This issue should be resolved at some point though. 

Even though you don't have access to the user object, you should still be able to make API calls to tables with row level security because the auth tokens are available.

**Reply by sumchans** [reply]
Okay thanks Tyler.

**Reply by sumchans** [reply]
@dshukertjr In the supabase Flutter quickstart tutorial, How did you manage to persist the session of the user, means after signup and signin, if the user quits and relaunches the app, how do we get the user to navigate into the app without going through the signup or signin process again. 

I see that it is all handled in the auth_state.dart file. Currently when starting the app even if the user was signed in before it will still ask you to signin and I see the logs saying current session data is missing. Here is the link to the code https://github.com/sumchans/Bol . Again I have been only working on the login_page.dart, all the other files are left untouched.

**Reply by dshukertjr** [reply]
I'm assuming your session was not persisted after signing up. Try sining in and see if it's persisted. It was persisted for me. 

So there is this bug currently that is preventing sessions from being persisted if you are just signing up and verifying OTP. 
>There seems to be a bug on the gotrue server where user object is not returned upon successful OTP verification. https://github.com/supabase/gotrue/issues/141

With phone number sign in, in order to persist session today, you need to sign in with phone number and password. Like this:
```dart
final response = await supabase.auth.signin(phone: phoneNumber, password:  '123456');
```

I saw that you are not calling this within your`_verifyOtp ()` method. That is the reason why your session upon signing up is not being persisted. 

If I were you though, I would just ignore the session persistence issue and wait for the gotrue issue to be solved. Once that is solved, you will be able to persist session without using password.

**Reply by sumchans** [reply]
Right session is not persisted. If I do sign in with the password after verifyOtp, I get the unsupported email provider message. Not sure why I am getting all these messages. As you said it is better to wait for gotrue to fix it so that I can have a clean looking code.
